Galaxus.Functional.Tests/(Option)/OptionTests.cs

```diff
@@ -143,6 +144,110 @@ public void Option_MatchThrowsIfMatchArmIsNull()
         Assert.Throws<ArgumentNullException>(() => { _ = Option<int>.None.Match(v => v, null); });
     }
 
+    [Test]
+    public async Task Option_MatchAsyncSome_Works()
+    {
+        var some = "hello".ToOption();
+
+        {
+            var called = false;
+            await some.MatchAsync(async _ =>
+            {
+                called = true;
+                await Task.CompletedTask;
+            }, async () =>
+            {
+                Assert.Fail();
+                await Task.CompletedTask;
+            });
+            Assert.IsTrue(called);
+        }
+
+        {
+            var called = false;
+
+            var number = await some.MatchAsync(async _ =>
+                {
+                    called = true;
+                    await Task.CompletedTask;
+                    return 42;
+                },
+                () =>
+                {
+                    Assert.Fail();
+                    throw new InvalidOperationException();
+                });
+
+            Assert.AreEqual(42, number);
+            Assert.IsTrue(called);
+        }
+
+        {
+            var called = false;
+            some.IfSome(_ => called = true);
+            Assert.IsTrue(called);
+        }
+    }
+
+    [Test]
+    public async Task Option_MatchAsyncNone_Works()
+    {
+        var none = Option<string>.None;
+
+        {
+            var called = false;
+            await none.MatchAsync(async _ =>
+            {
+                Assert.Fail();
+                await Task.CompletedTask;
+            }, async () =>
+            {
+                called = true;
+                await Task.CompletedTask;
+            });
+            Assert.IsTrue(called);
+        }
+
+        {
+            var called = false;
+
+            var number = await none.MatchAsync(_ =>
+                {
+                    Assert.Fail();
+                    throw new InvalidOperationException();
+                },
+                async () =>
+                {
+                    called = true;
+                    await Task.CompletedTask;
+                    return 42;
+                });
+
+            Assert.AreEqual(42, number);
+            Assert.IsTrue(called);
+        }
+
+        {
+            var called = false;
+            none.IfSome(_ => called = true);
+            Assert.IsFalse(called);
+        }
+    }
+
+    [Test]
+    public void Option_MatchAsyncThrowsIfMatchArmIsNull()
+    {
+        // SOME
+        Assert.ThrowsAsync<ArgumentNullException>(async () => { await 0.ToOption().MatchAsync(null, async () => { await Task.CompletedTask; }); });
+
+        // NONE
+        Assert.ThrowsAsync<ArgumentNullException>(async () => { await Option<int>.None.MatchAsync(async _ => { await Task.CompletedTask; }, null); });
+
+        // SOME and NONE with return value
+        Assert.ThrowsAsync<ArgumentNullException>(async () => {_ = await 0.ToOption().MatchAsync(null, () => Task.FromResult(0)); });
```

Comment:
It's rather new, so you haven't had access to it before, but in the latest master there's an `OptionFactory` (and a `ResultFactory`) in the test project. You could replace the `0.ToOption()` and similar with `CreateSome()` and `CreateNone()` - though they only exist in their async variants for now.

Reply:
Thank you! I don't see the `CreateSome()` method, just `CreateSomeTask()`. Is that the one you meant?

Reply:
In a way - for `Option`, there's just `CreateSomeTask` and `CreateNoneTask` right now, but since we're gonna need them, you can add `CreateSome` and `CreateNone` and use them.

In `ResultFactory`, this is done already - both task-wrapped and "bare" versions exist. So you can look there for inspiration! ;)

Reply:
Added and used `CreateSome`